pennylane/ops/functions/matrix.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the qml.matrix function.
+"""
+# pylint: disable=protected-access
+import pennylane as qml
+
+
+@qml.op_transform
+def matrix(op, *, wire_order=None):
+    r"""The matrix representation of an operation or quantum circuit.
+
+    Args:
+        op (.Operator, pennylane.QNode, .QuantumTape, or Callable): An operator, quantum node, tape,
+            or function that applies quantum operations.
+        wire_order (Sequence[Any], optional): Order of the wires in the quantum circuit.
+            Defaults to the order in which the wires appear in the quantum function.
+
+    Returns:
+        tensor_like or function: Function which accepts the same arguments as the QNode or quantum
+        function. When called, this function will return the unitary matrix in the appropriate
+        autodiff framework (Autograd, TensorFlow, PyTorch, JAX) given its parameters.
+
+    **Example**
+
+    Given an operation, ``qml.matrix`` returns the matrix representation:
+
+    >>> op = qml.RX(0.54, wires=0)
+    >>> qml.matrix(op)
+    [[0.9637709+0.j         0.       -0.26673144j]
+    [0.       -0.26673144j 0.9637709+0.j        ]]
+
+    It can also be used in a functional form:

```

Comment:
```suggestion
    It can also be used in a functional form when applied to an operator subclass:
```

Reply:
ah I didn't want to say this, because it also works for arbitrary functions, not just operator subclasses! but any callable

Reply:
Oh of course